### PR TITLE
Dataset carinalities experiments

### PR DESCRIPTION
**Pull request by kris701**
* Added experiments for how well the CGAN does with unbalanced data (Accuracy and Quality)
* Put the `CGAN` and `Classifier` ML models into seperate modules, so that they can actually be overwritten in experiments
* Made a fix so that the letter producer wont crash when trying to make more than 1000 images

### CODE REVIEW

File: JANGAN/CGAN/LetterProducer.py [hunk 1]
```diff
@@ -23,6 +23,22 @@ def __init__(self, outputPath, trainedGenerator, numberOfClasses, latentDimensio
         self.ImageCountToProduce = imageCountToProduce
 
     def GenerateLetter(self, classID, imageCountToProduce):
+        returnImages = None
+        if imageCountToProduce > 500:
```
Comment (jamadaha) [inline JANGAN/CGAN/LetterProducer.py:27]: Please don't use magic numbers.

**Review by jamadaha — CHANGES_REQUESTED** [review]

File: JANGAN/ExperimentQueue/BaseRunConfig.ini [hunk 2]
```diff
@@ -173,3 +171,4 @@ FormatImages = True
 LogDir = "../../Data/{EXPERIMENTNAME}/{TIMESTAMP}/Logs/"
 # Path of the classification images
 ClassifyDir = "../../Data/{EXPERIMENTNAME}/{TIMESTAMP}/FakeOutput/Final/"
+FormatImages = False
```
Comment (jamadaha) [inline JANGAN/ExperimentQueue/BaseRunConfig.ini:174]: Please add comment of what this does

**Review by jamadaha — DISMISSED** [review]

**Review by HGinnerup — CHANGES_REQUESTED** [review]

File: JANGAN/ExperimentQueue/Experiment3/UnbalancedDatasetQuality.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+class newCGANTrainer(cgt.CGANTrainer):
+    CSVPath = ""
+    CSVData = {}
+    CSVDataLabels = {}
+    HighestValue = 0
+
+    def __init__(self, model, datasets, epochs, refreshUIEachXStep, saveCheckPoints, checkpointPath, latestCheckpointPath, logPath, numberOfClasses, latentDimension, epochImgDir):
+        super().__init__(model, datasets, epochs, refreshUIEachXStep, saveCheckPoints, checkpointPath, latestCheckpointPath, logPath, numberOfClasses, latentDimension, epochImgDir)
+        self.CSVPath = "../../Data/Distribution.csv"
```
[the remark this answers is not in the document]
Comment (kris701) [reply JANGAN/ExperimentQueue/Experiment3/UnbalancedDatasetQuality.py:35]: It would simply be too much effort to get that config value in here for its worth

**Review by HGinnerup — APPROVED** [review]
